bolt/organization.go

```diff
@@ -311,16 +327,30 @@ func organizationIndexKey(n string) []byte {
 }
 
 // forEachOrganization will iterate through all organizations while fn returns true.
-func forEachOrganization(ctx context.Context, tx *bolt.Tx, fn func(*influxdb.Organization) bool) error {
+func forEachOrganization(ctx context.Context, tx *bolt.Tx, opts influxdb.FindOptions, fn func(*influxdb.Organization) bool) error {
 	cur := tx.Bucket(organizationBucket).Cursor()
-	for k, v := cur.First(); k != nil; k, v = cur.Next() {
+
+	var k, v []byte
+	if opts.Descending {
```

Comment:
One this I did to avoid this conditional check was to just set the functions for `Next/Prev`

https://github.com/influxdata/influxdb/blob/4d5f39334e08f46ee37130aa6155c4125f0ac71c/bolt/keyvalue_log.go#L182-L191

Not sure how I feel about it, but it does avoid the conditional logic in the core loop.

Reply:
I think to like it! Will look better in some minutes when I'll be back at my laptop. Thanks!